packet-ethereum-disc.c

```diff
@@ -693,20 +995,32 @@ static gboolean dissect_ethereum_heur(tvbuff_t *tvb, packet_info *pinfo, proto_t
     return FALSE;
   }
 
-  guint packet_type = tvb_get_guint8(tvb, ETHEREUM_DISC_PACKET_TYPE_IDX);
-  if (packet_type < UNKNOWN || packet_type > NODES) {
-    return FALSE;
-  }
+  // https://github.com/ethereum/go-ethereum/blob/c4712bf96bc1bae4a5ad4600e9719e4a74bde7d5/p2p/discv5/udp.go#L149
+  static const gchar *discv5_id = "temporary discovery v5";
+  gboolean is_discv5 = FALSE;
+  const gchar *version_prefix = tvb_get_string_enc(wmem_packet_scope(), tvb, 0, strlen(discv5_id), ENC_ASCII);
+  if (strcmp(version_prefix, discv5_id) == 0) {
+      is_discv5 = TRUE;
+  } else {
+      guint packet_type = tvb_get_guint8(tvb, ETHEREUM_DISC_PACKET_TYPE_IDX);
+      if (packet_type > NODES) {
```

Comment:
This check would abort the dissection too early, and would preclude the `(Unknown packet ID: %d)` message from showing, which can be useful to debug unsupported messages that contain what appears to be RLP.

Reply:
I didn't add this code, though -- just moved it to the else block as it doesn't make sense when we know it's a v5 packet. Should I remove it altogether?